contracts/minter-suite/Minters/MinterDAExpRefund/MinterDAExpRefundV0.sol

```diff
@@ -0,0 +1,997 @@
+            // do not split revenue here since will be claimed at a later time.
+            _projectConfig.numRefundableInvocations++;
+        }
+
+        return tokenId;
+    }
+
+    /**
+     * @notice Refunds the sender's payment above current amount due for
+     * project `_projectId`. The current amount due is the the price paid for
+     * the most recently purchased token, or the base price if the artist has
+     * withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auction has sold out above base price or after the auction
+     * has been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Sends refund to msg.sender.
+     * @param _projectId Project ID to refund payment on.
+     */
+    function claimRefund(uint256 _projectId) external {
+        claimRefundTo(payable(msg.sender), _projectId);
+    }
+
+    /**
+     * @notice Refunds the sender's payment above current amount due for
+     * project `_projectId`. The current amount due is the the price paid for
+     * the most recently purchased token, or the base price if the artist has
+     * withdrawn revenues after the auction reached base price.
+     * This function is callable at any point, but is expected to typically be
+     * called after auction has sold out above base price or after the auction
+     * has been purchased at base price. This minimizes the amount of gas
+     * required to refund the sender.
+     * Sends refund to address `_to`.
+     * @param _to Address to send refund to.
+     * @param _projectId Project ID to refund payment on.
+     */
+    function claimRefundTo(address payable _to, uint256 _projectId)
+        public
+        nonReentrant
+    {
+        ProjectConfig storage _projectConfig = projectConfig[_projectId];
+        Receipt storage receipt = receipts[msg.sender][_projectId];
+        // CHECKS
+        // input validation
+        require(_to != address(0), "No refund to the zero address");
+        // require that a user has purchased at least one token on this project
+        require(receipt.numPurchased > 0, "No purchases made by this address");
+        // get the latestPurchasePrice, which returns the sellout price if the
+        // auction sold out before reaching base price, or returns the base
+        // price if auction has reached base price and artist has withdrawn
+        // revenues.
+        // @dev if user is elligible for a refund, they have purchased a token,
+        // therefore we are guaranteed to not have a populated
+        // latestPurchasePrice
+        uint256 netTokenPrice = _projectConfig.latestPurchasePrice;
+
+        // EFFECTS
+        // calculate the refund amount
+        // implicit overflow/underflow checks in solidity ^0.8
+        uint256 amountDue = receipt.numPurchased * netTokenPrice;
```

Comment:
Nit: would something like "netPaidAfterRefund" or "baseCost" be more accurate here vs. "amountDue" maybe?

Reply:
hmm amountDue makes the most sense in my head, but I could be living in my own world here 😆 @mchrupcala and @jakerockland - any other thoughts on variable name here?

Reply:
Ah 😅 I think I see it -- so "amount due" reads to me like it's owed to the user (and so is the refund), which is why it read strangely to me when subtracting `amountDue` to calculate `refund`. But this `amountDue` is actually owed to a split of the artist/artist payee/AB. It's a very minor nit but what about `amountDueToPayees` or something similar?

Reply:
good call, I see how that is confusing! I've pushed an update to change from `amountDue` to `requiredAmountPosted` in c94d151. That kind of jives with `receipt.netPosted` I think?

Reply:
resolving, and we can iterate on this going forward!